### ISSUE TITLE
Echo middleware incompatible with v4

### ISSUE THREAD

**Issue by noamt** [honeycombio/beeline-go#69]
The latest version of Echo [requires](https://github.com/labstack/echo#supported-go-versions) the you explicitly import the module using the `/v4` prefix:

> Any of these versions will allow you to import Echo as github.com/labstack/echo/v4 which is the recommended way of using Echo going forward.

So the current Echo middleware is usable only with version 3:
```go
./main.go:56:33: cannot use hnyecho.New().Middleware() (type "github.com/labstack/echo".MiddlewareFunc) as type "github.com/labstack/echo/v4".MiddlewareFunc in argument to e.Use
```

 In the meantime I'm able to work around this issue by duplicating the code from the middleware into my own middleware that imports `/v4`

**Comment by wolfeidau** [comment]
This looks to be related to the gomod support, it would be good to get this fixed as most new tools and libraries use echo v4.

**Comment by vreynolds** [comment]
Closed by #99
